metron-platform/metron-common/src/main/java/org/apache/metron/common/query/PredicateProcessor.java

```diff
@@ -18,12 +18,20 @@
 
 package org.apache.metron.common.query;
 
-import org.antlr.v4.runtime.*;
-import org.apache.metron.common.query.generated.*;
+import org.antlr.v4.runtime.ANTLRInputStream;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.TokenStream;
+import org.apache.metron.common.query.generated.PredicateLexer;
+import org.apache.metron.common.query.generated.PredicateParser;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
 
 
 public class PredicateProcessor {
   public boolean parse(String rule, VariableResolver resolver) {
+    if (isEmpty(rule)) {
```

Comment:
On second thought, what about rule == null || isEmpty(rule.trim())? I can't think of a reason to handle null different from "" or " ". The trim would only be evaluated if rule != null, and now we're also covering the null condition. Thoughts?